Adds 33 new get and post tests.

markcherbaka opened the pull request.

markcherbaka committed:
Adds 33 new get and post tests.

dlongley commented on lib/config.js:
```diff
@@ -7,7 +7,10 @@ var config = require('bedrock').config;
+// var basePath = '/keys';
+// config['key-http'] = {};
+// config['key-http'].routes = {};
+// config['key-http'].routes.basePath = basePath;
```
What's going on here with the config being removed and commented out?

markcherbaka replied:
bedrock-key-http's main.js doesn't use the config in this way... perhaps it needs to be updated?  Anyway, the basePath gets set here based on other bedrock config, not the config.js file.  I believe these were commented out because they weren't used.  

Here's where basePath gets set in main:  https://github.com/digitalbazaar/bedrock-key-http/blob/master/lib/main.js#L34
Here's where that configuration gets set in bedrock-key:  https://github.com/digitalbazaar/bedrock-key/blob/testFramework/lib/config.js#L13

If we are trying to standardize how we do this, we can update main.js.  For example, the path in bedrock-agreement-http gets set using it's own config like the one commented out above here:  https://github.com/digitalbazaar/bedrock-agreement-http/blob/master/lib/index.js#L24

You or Matt might have more insight on this.

dlongley replied:
Sounds like we should make `basePath` here a computed config variable based on bedrock-key's config var. @davidlehn ?

dlongley replied:
Also, `main.js` should be renamed to `index.js` to follow established practice.

davidlehn replied:
I don't know why basePath is needed in bedrock-key-http if it's in bedrock-key.  What is the usefulness of aliasing the var to a different namespace?  I don't think we need to leave the commented code in the file.  Is there a policy for adding an empty config or just omit it?

markcherbaka replied:
I might need Matt's input on this, but in general, I think the path (called basePath here, but it should be renamed keyPath) does need to be made into a computed config variable.  I'll talk to Matt about it next week and resolve this.  I'll either update, or make a note in a comment on why that commented code is there.

dlongley commented on test/package.json:
```diff
@@ -0,0 +1,49 @@
+  "directories": {
+    "lib": "./lib"
+  }
+}
```
Needs EOL

dlongley commented:
I just have a couple of comments to address here, otherwise LGTM. I think those indent changes are unnecessary, but I recall a conversation @mattcollier and I had about that particular style. I'm not for/against either way of doing it at the moment, so it's fine if we change it or leave it be here.

davidlehn commented:
I'll weigh in on the indent issue, pardon the distraction from the meat of the PR ;-)

I'd rather not see extra indents like that.  It's confusing since it's an extra level deep not because of program structure but because of handling a lone line.

An option is double indenting the continued line to make it clear they are not at the same level as the block content.  I like that style but I know some people don't:
```js
var doubleIndentContinuation = function(
    arg1, arg2, ...) {
  body();
}
```

Compare to:
```js
var doubleIndentContent = function(
  arg1, arg2, ...) {
    body();
}
```
```js
var noIndent = function(
  arg1, arg2, ...) {
  body();
}
```

markcherbaka commented:
davidlehn, I like your doubleIndentContinuation; however, I think we need to agree on something before I rip up a bunch of formatting.  Also, I find it extremely convenient to have a linter that works with whatever we decide.  For now, that seems to be the doubleIndentContent.  I really don't like the noIndent option, but I'm happy to go with whatever you all agree to.

markcherbaka commented:
Any feedback on this?  I believe it's still open and in Dave and Matt's court.

dlongley commented:
I think we should stick with `noIndent` from @davidlehn's example above. We've used that pattern in most of the code (I think). Let's not change it up now. After the indent stuff is handled, @davidlehn, do you give your blessing to get this PR merged?

markcherbaka commented:
There are several small, but open, issues with this pull, and I will attempt to resolve them next week:
1.) Computing 'basePath'.
2.) Renaming of main.js
3.) Comments:  We agreed not to update what's there, but we will be doing the 'noIndent' option moving forward, with an option to going to a doubleIntent pending further discussion.
4.) Dave Lehn brought up that he doesn't know how to run these tests.  I am happy to write up a short guide for this.  Will discuss w/ Collier where that should go, since it does apply to more than just this module.

davidlehn commented on lib/index.js:
```diff
@@ -31,15 +31,16 @@ bedrock.events.on('bedrock-express.init', function(app) {
 function addRoutes(app) {
-  var basePath = bedrock.config.key.basePath;
+  var keyPath = bedrock.config['key-http'].routes.basePath;
+  console.log('basePath is :', keyPath);
```
remove logging

markcherbaka commented:
Just for the record, I've addressed all the concerns that have come up:
1.) 'basePath' computing has been added
2.) main.js has been renamed to index.js
3.) comments required no action
4.) A short guide for testing is being developed under bedrock-test

I believe this is ready to be merged next time one of you gets a chance.  Let me know if there are any outstanding issues or questions.

davidlehn commented:
Should probably add `"private":true` to `test/package.json` since that's not something that will be published.

mattcollier committed:
Code clean-up.

mattcollier committed:
Add CI badge.
